### ISSUE TITLE
[Proposal] Using ECE Tools with Magento Open Source

### ISSUE THREAD

**Issue by ihor-sviziev** [magento/ece-tools#512]
As ECE Tools became open source and public available - it's good idea to add instructions how to use it with Magento OS. As result - people might try to use it on Magento OS and will report issues / send PRs.
I just tried and got few issues:

- [ ] all docs are saying how to use it on Magento Commerce, but none are saying how to use them with Magento OS
- [x] package magento/ece-tools is not public available, even not available when I put my credentials w/o access to Magneto Commerce, there only one way to install it - use link to github repo as a composer repository

**Comment by YPyltiai** [comment]
Hi @ihor-sviziev, 

Thanks for creating the proposal.

- I believe that's an error that the package is not available with non-EE keys now. Assume that's because some time ago it was private and we didn't change entitlements on composer repo to align with GitHub. We'll look into fixing this part.
- For a bigger question of OS support, we'll need some time to review this request. We never really tested it thoroughly to work with OS only, without any EE modules or Cloud meta-package components. Even though we believe that it should work fine even now, having ECE-Tools officially OS-compatible was not on the roadmap before.

Do you see much interest from OS community in ECE-Tools?
Were you able to get it working after pulling package itself from GitHub?

**Comment by YPyltiai** [comment]
@ihor-sviziev , quick update: latest version `2002.0.19` is available with OS keys now.
Still working on validation of the main proposal though.

**Comment by aepod** [comment]
I was able to get this working fairly well on OS version, just by including the package with composer require. I had an auth key set of course, but with no EE permissions.

I think the community would really benefit from having a single source of truth when it comes to build/deploy/docker etc. This CI/CD tool can easily fill the gap, and it also brings more people into the cloud fold by teaching them how to be compatible with cloud. It will be a lot easier to acclimate if they ever go from Open Source to Cloud Edition.

Another very big benefit is that the ece-tools package may get some very nice pull requests that include features or fixes, a lot of the other tools benefit from this already.

Just throwing my 2-cents in here, because I was literally seeing how the water felt and was trying to write up a howto etc on using ece-tools for build/deploy/docker if you are not on cloud.

**Comment by IJOL** [comment]
I will be a happy user of ece-tools with the OS version of magento, in addition to fill the gap now, Ii wil be grateful for ever if some one  graciously put a detailed tutorial on how to use ece-tools local docker deployment with Magento OS..

**Comment by YPyltiai** [comment]
@shiftedreality , @aepod ^, I think you both managed to install OS with ECE-Tools on Cloud Docker?

**Comment by shiftedreality** [comment]
Yes, we have automated Functional build which also verifies this functionality. Should be closed with 2002.1 release.

**Comment by IJOL** [comment]
Hola a todos, I will be glad to test anything you come up with, but i admit
i dont have my composer skills up to the task, i'¡m very lost about how to
that, a detailed guidance will be greatly appreciated..


Saludos,
--------------------------------------------
Ignacio J.Ortega
skype:ignacioj.ortega

**Comment by vpashovski** [comment]
It will be great to use ece-tools to build CI/CD for Magento Open Source. As I can see most of the steps in hooks are related with MagentoCloud module.
Is it there easy way to use it for that?
Or we have to remove cloud dependencies and build all steps again?

**Comment by shiftedreality** [comment]
@vpashovski there are few options available:

1. Use Magento Cloud-like way of deployment, with `build`, `deploy` and `post-deploy` phases. It does not really require Magento Cloud. We can potentially use it on any infrastructure with small tweaks.
2. Write custom scenarios: https://devdocs.magento.com/cloud/deploy/scenario-based-deployment.html#add-scenarios-using-build-and-deploy-hooks

Overall ECE-Tools is already compatible with Magento OpenSource, we'll most likely add some topic on DevDocs.

cc: @meker12

**Comment by shiftedreality** [comment]
@ihor-sviziev we can cross-out this:

> package magento/ece-tools is not public available, even not available when I put my credentials w/o access to Magneto Commerce, there only one way to install it - use link to github repo as a composer repository

ECE-Tools is available with OS keys now

**Comment by ihor-sviziev** [comment]
@shiftedreality thank you for update! I updated issue description.
But still some example is missing.

Would be great to have something similar to https://github.com/fooman/magento2-starter-template/runs/538326475?check_suite_focus=true, so many people could try it

**Comment by NadiyaS** [comment]
@magento export issue to Jira project MCLOUD as Task

**Comment by NadiyaS** [comment]
@magento export issue to Jira project MCLOUD as Task

**Comment by NadiyaS** [comment]
@magento export issue to Jira project MCLOUD as Task

**Comment by magento-engcom-team** [comment]
:white_check_mark: Jira issue (https://jira.corp.magento.com/browse/MCLOUD-7179) is successfully created for this issue.

**Comment by ihor-sviziev** [comment]
@NadiyaS, any info on why this issue was closed?

**Comment by shiftedreality** [comment]
@ihor-sviziev we have another work going around Git installation from OpenSource (https://www.youtube.com/watch?v=P6lbHZa-uPo&feature=youtu.be) and wrongly assumed it duplicates this ticket.

I've re-opened this ticket for Composer-based installation of Magento OS
